polkadot/xcm/xcm-executor/src/traits/weight.rs

```diff
@@ -22,11 +22,11 @@ use xcm::latest::{prelude::*, Weight};
 pub trait WeightBounds<RuntimeCall> {
 	/// Return the maximum amount of weight that an attempted execution of this message could
 	/// consume.
-	fn weight(message: &mut Xcm<RuntimeCall>) -> Result<Weight, ()>;
+	fn weight(message: &mut Xcm<RuntimeCall>) -> Result<Weight, XcmError>;
```

Comment:
Can we make this return the index of the problematic instruction as well? It'd be very useful to bubble it up in https://github.com/paritytech/polkadot-sdk/pull/7730